crates/nu-command/src/network/fetch.rs

```diff
@@ -361,6 +361,22 @@ fn response_to_buffer(
     engine_state: &EngineState,
     span: Span,
 ) -> nu_protocol::PipelineData {
+    // Try to get the size of the file to be downloaded.
+    // This is helpful to show the progress of the stream.
+    let buffer_size = match &response.headers().get("content-length") {
+        Some(content_length) => {
+            let tmp_header = &(*content_length).clone(); // binding
+
+            Some(
+                tmp_header
+                    .to_str()
+                    .unwrap_or("")
+                    .parse::<u64>()
+                    .unwrap_or(0),
+            )
```

Comment:
```suggestion
            let content_length = content_length
            .to_str()
            .unwrap_or("")
            .parse::<u64>()
            .unwrap_or(0);

            if content_length == 0 {
                None
            } else {
                Some(content_length)
            }
```

I think it would be better to return `None` if we can't parse the `content-length` header. Otherwise the progress bar looks like it's at 100% the entire time.